### PR TITLE
support for differently named identifier columns

### PR DESCRIPTION
**Pull request by hakanesnn**
## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [ ] My code fits the style guidelines.
- [ ] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by Manason** [comment]
I'm not sure the database schema should be something open to be dynamically configured. Why are you desiring to rename the id column?

**Comment by Manason** [comment]
> When I install with qbox txadminrecipe, renewed-banking is installed, renewed-banking creates the created accounts to the db with the creator column, not the citizenid column, I think that when the character is deleted, the accounts belonging to the character should also be deleted. this is just an example of renewed-banking, other script creators may not save the data to a column named citizenid when saving to the db, I needed something like this because the characters are now managed from within qbx-core.

Could we solve this problem by triggering an event when player is deleted and then you could clean up non qbx tables by consuming that event?

File: server/storage.lua [hunk 1]
```diff
@@ -225,28 +225,12 @@ end
...
-        'playerskins',
-        'player_contacts',
-        'player_houses',
-        'player_mails',
-        'player_outfits',
-        'player_vehicles',
-    }
-
-    local query = "DELETE FROM %s WHERE citizenid = ?"
+    local query = "DELETE FROM %s WHERE %s = ?"
```
Comment (FjamZoo) [inline server/storage.lua:228]:
```suggestion
    local query = 'DELETE FROM %s WHERE %s = ?'
```

pretty please thank you

**Comment by FjamZoo** [comment]
> Could we solve this problem by triggering an event when player is deleted and then you could clean up non qbx tables by consuming that event?

or imagine this we could use something thats called foreign keys 🤯

File: config.lua [hunk 2]
```diff
@@ -109,4 +109,21 @@ QBConfig.Server.Discord = "" -- Discord invite link
...
+    ['phone_invoices'] = 'citizenid',
+    ['phone_messages'] = 'citizenid',
+    ['playerskins'] = 'citizenid',
+    ['player_contacts'] = 'citizenid',
+    ['player_houses'] = 'citizenid',
+    ['player_mails'] = 'citizenid',
+    ['player_outfits'] = 'citizenid',
+    ['player_vehicles'] = 'citizenid',
+    -- ['bank_accounts_new'] = 'creator'
+}
```
Comment (Manason) [inline config.lua:127]: Could we put this in a new Config object? I'm not sure there is a reason to expose this table data outside of the core.

Also consider putting a comment that these are the rows that will be deleted when a character is deleted.

Also consider renaming to CharacterDataTables. Using character as much as possible will open us up to one day separating the concept of a player

**Review by Manason — CHANGES_REQUESTED** [review]

**Comment by Manason** [comment]
> Yes it solves but I don't think it will be user friendly. As I understand you don't want to write non qbx tables into qbx-core, you can do it by leaving only qbx tables in the cfg I sent. Users who want to add other tables can configure the config file.

Ok. I think I'm convinced since we do want everything in one transaction. See my requested changes, but I'm willing to approve the concept of this PR.

File: config.lua [hunk 3]
```diff
@@ -110,3 +110,21 @@ QBConfig.Server.CheckDuplicateLicense = true -- Check for duplicate rockstar lic
 QBConfig.Server.Permissions = { 'god', 'admin', 'mod' } -- Add as many groups as you want here after creating them in your server.cfg
 
 QBConfig.NotifyPosition = 'top-right' -- 'top' | 'top-right' | 'top-left' | 'bottom' | 'bottom-right' | 'bottom-left'
+
+---@alias TableName string
+---@alias ColumnName string
+---@type table<TableName, ColumnName>
+CharacterDataTables = {
```
Comment (Manason) [inline config.lua:117]: Config = {}
Config.CharacterDataTables = ...

I like having the Config object to signify that this isn't supposed to be a mutable global variable, but rather an immutable field set in config

**Review by Manason — CHANGES_REQUESTED** [review]

**Review by Manason — APPROVED** [review]

**Pull request merged**